pysensors/utils/_validation.py

```diff
@@ -0,0 +1,48 @@
+"""
+Various utility functions for validation and computing reconstruction scores and errors.
+"""
+import numpy as np
+
+def determinant(top_sensors, n_features, basis_matrix):
+    """
+    Function for calculating |C.T phi.T C phi|.
+
+    Parameters
+        ----------
+        top_sensors: np.darray,
+            Column indices of choosen sensor locations
+        n_features : int,
+            No. of features of dataset
+        basis_matrix : np.darray,
+            The basis matrix calculated by model.basis_matrix_
+        
+        Returns
+        -------
+        optimality : Float, 
+            The dterminant value obtained.
+    """
+    
+    c = np.zeros([len(top_sensors),n_features])
+    for i in range(len(top_sensors)):
+        c[i,top_sensors[i]] = 1
+    phi = basis_matrix
```

Comment:
I think the basis_matrix is not the phi we are looking for. I will try to fix this.

Reply:
I have not touched _validation.py
This is when I pulled your work, if you have updated this since then, please feel free to correct what you see necessary.